torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1243,14 +1255,12 @@ std::pair<std::vector<Val*>, bool> Index::getConsumerRootPredIndices(
   // If we are generating a predicate for initialization check if we should use
   // rfactor instead of root_dom
   bool use_rfactor = true;
-  if (consumer_tv->hasRFactor()) {
-    auto rfactor_dom = consumer_tv->getMaybeRFactorDomain();
+  if (consumer_tv->domain()->hasRFactor()) {
```

Comment:
Can we have `hasRFactor` in `kir::TensorView` as well?

Reply:
This is another example of the "shallow module" anti-pattern: it would explode the `kir::TensorView` interface for a questionable convenience. 

`tv->domain()->hasRfactor()` makes it clear that the domain has the "rfactor" information.